src/isl/isl_type_reference.rs

```diff
@@ -47,13 +72,25 @@ impl IslTypeRef {
                         "a base or alias type reference can not be null.symbol",
                     )
                 }
-                value.as_sym().unwrap()
+
+                let type_name = value.as_sym().unwrap()
                     .text()
                     .ok_or_else(|| {
                         invalid_schema_error_raw(
                             "a base or alias type reference symbol doesn't have text",
                         )
-                    }).map(|type_name| IslTypeRef::Named(type_name.to_owned()))
+                    })?;
+
+                // check for nullable type reference
+                if value.annotations().any(|a| a == &text_token("nullable")) {
```

Comment:
I think this allocates a new `String` (`"nullable"`) on each pass, which is expensive. Instead, see if `a`'s text matches `"nullable"` (the `&str`).

Reply:
Maybe another reason to implement https://github.com/amzn/ion-rust/issues/400?

Reply:
I agree. I think it can be used on many places in `ion-schema-rust`. I will create a PR for implementation of `has_annotation()`  along with `len()` for container type of `OwnedElement`s  because that is also used multiple times in `ion-schema-rust`.